### PR TITLE
cmd, ci, docs: Add test script for get-started.md

### PR DESCRIPTION
**Pull request by patzielinski**
This PR adds a test script to test the commands in `docs/get-started.md`, as suggested in #394. (Thanks @lukpueh!)

At the moment, this checks that the return codes of the commands tested are all `0`'s, and raises an error if a non-zero code is returned. I can rework this to match parts of the output if we think that's better/needed (parts since output is variable in places).

### CODE REVIEW

**Comment by neilnaveen** [comment]
Don't we already have somthing like this in the gittuf demo repo?

**Comment by adityasaky** [comment]
cc @lukpueh

**Review by lukpueh — COMMENTED** [review]
This also works, but requires manual syncing with the markdown file. May I ask, why you decided against scraping and running the actual snippets from the markdown as suggested in #394?

**Comment by lukpueh** [comment]
> any change in commands or command output requires updating the script with the expected text anyways.

Sure, but CI will fail if the actual output doesn't match the expected output anymore.

**Comment by lukpueh** [comment]
> Maybe we can add an expected output text file that the script loads and compares to the terminal output so we can just update that file instead and have the script extract commands from the markdown file.

Yes, that's what the script referenced in the issue does.

**Comment by adityasaky** [comment]
timestamps in the debug messages may be a small blocker for that? Perhaps this is a good time to pretty that up as well, though I think it's low priority...

**Comment by lukpueh** [comment]
> This is what I meant about variable output, as the timestamps prevent checking output equivalence. I can just disable the verbose output and match on the small amount of output we have anyways...

Ah right, it's harder if timestamps are in the output. There are several ways to go about this:
- change gittuf to not print timestamps
- run the test with something like `faketime`
- strip timestamps from the output in the tests


> A question though, is it better to have the output embedded in the script or as a separate file? The script referenced appears to do the former.

Whatever you prefer. If it's short enough, like in the in-toto demo, I think it's fine to just add it to the script. If it's a lot longer a separate file is probably better for readability of the script.

**Comment by adityasaky** [comment]
Can we make hashes deterministic by controlling the author/committer details (including the time) and signing with an ed25519 key? You can pin those details using env vars.

**Comment by lukpueh** [comment]
> > Can we make hashes deterministic by controlling the author/committer details (including the time) and signing with an ed25519 key? You can pin those details using env vars.
> 
> I can try that, but do we want to update the commands in the markdown to match? This should be to test the commands as the user would run them with only minor changes imo...

Setting up custom author/committer details and a fake time shouldn't affect the testable snippets. Key type maybe, but I guess that's okay. 

Btw. after #414, rsa keys will produce reproducible signatures too, because we'll no longer sign ourselves (with PSS), but ask `ssh-keygen` to sign, which uses (pkcsv1.5).

File: docs/testing/tester-expected-macos.txt
Comment (lukpueh) [inline docs/testing/tester-expected-macos.txt:23]: I think the single quotes being added on set -x is a bash thing. Maybe, if you explicitly run the script with bash, you could use the same expected output un ubuntu and macOS.

File: .github/workflows/get-started-tests.yml
Comment (adityasaky) [inline .github/workflows/get-started-tests.yml:25]: please pin hash, thanks

File: docs/testing/test-get-started-md.py
Comment (adityasaky) [inline docs/testing/test-get-started-md.py:70]: can this be ISO 8601 / rfc 3339?

File: docs/testing/test-get-started-md.py
Comment (adityasaky) [inline docs/testing/test-get-started-md.py:72]: https://github.com/gittuf/gittuf/blob/main/internal/gitinterface/common.go#L17-L20 for consistency, let's use these values? :D

**Comment by adityasaky** [comment]
@lukpueh could you take a look and approve if this is good to go? thanks!

File: docs/testing/test-get-started-md.py
Comment (lukpueh) [inline docs/testing/test-get-started-md.py:37]: minor nit:

```suggestion
        case "Linux" | "Darwin":
            expected_output_file = os.path.realpath(os.path.join(testing_path, EXPECTED_OUTPUT_FILENAME))
```

**Review by lukpueh — APPROVED** [review]
Nice! look great

File: docs/testing/test-get-started-md.py
Comment (adityasaky) [inline docs/testing/test-get-started-md.py:46]: os.path.join(tmp_dir, "repo")?

**Review by adityasaky — APPROVED** [review]

**Comment by adityasaky** [comment]
Thanks @patzielinski!

**Pull request merged**
